IaC/lamda_module_template.tf

```diff
@@ -0,0 +1,149 @@
+resource "aws_iam_role" "LambdaRole" {
+  name = var.aws_iam_role_name
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "LambdaPolicy" {
+  name        = var.aws_iam_policy_name
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "ec2:CreateNetworkInterface",
+          "ec2:DescribeNetworkInterfaces",
+          "ec2:DeleteNetworkInterface"
+        ]
+        Effect   = "Allow"
+        Resource = "*"
+      },
+      {
+          "Effect": "Allow",
+          "Action": [
+              "logs:CreateLogStream",
+              "logs:PutLogEvents"
+          ],
+          "Resource": [
+              "${aws_cloudwatch_log_group.LambdaLogGroup.arn}:*"
+          ]
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy_attachment" "LambdaPolicyAttachment" {
+  name       = var.aws_iam_policy_attachment_name
+  roles      = [aws_iam_role.LambdaRole.name]
+  policy_arn = aws_iam_policy.LambdaPolicy.arn
+}
+
+resource "aws_cloudwatch_log_group" "LambdaLogGroup" {
+  name = var.aws_cloudwatch_log_group_name
+
+  retention_in_days = 7
+}
+
+resource "aws_lambda_function" "LambdaFunction" {
+    depends_on = [
+        aws_iam_policy_attachment.LambdaPolicyAttachment
+    ]
+
+  filename                       = var.file_name
+  function_name                  = var.function_name
+  role                           = aws_iam_role.LambdaRole.arn
+  handler                        = "bootstrap"
+  runtime                        = "provided.al2"
+  timeout                        = 60
+
+
+  
+  source_code_hash = filebase64sha256(local.file_name) 
+  vpc_config {
+    subnet_ids = [aws_subnet.MTBZoneLambdaSubnet.id]
```

Comment:
Both of these need to be `var`s on the module. This is to allow flexibility for choosing in which VPC to deploy the API.